# router.onError

Signature:

```js
router.onError(callback)
```

Register a callback which will be called when an error is caught during a route navigation. Note for an error to be called, it must be one of the following scenarios:

* The error is thrown synchronously inside a route guard function;

* The error is caught and asynchronously handled by calling next(err) inside a route guard function;

* An error occurred when trying to resolve an async component that is required to render a route
